src/traits/networking/centralized_server_network.rs

```diff
@@ -825,6 +830,8 @@ impl<M: NetworkMsg, K: SignatureKey + 'static, E: ElectionConfig + 'static> Conn
                     .context(FailedToDeserializeSnafu),
             }
         };
+        let _ =
+            <CentralizedServerNetwork<K, E> as ConnectedNetwork<M, K>>::shut_down::<'_, '_>(self);
```

Comment:
I don't think this will have the intended effect. When `recv_msgs` is called, `shut_down` will be called before the closure is evaluated. Is this what we want?